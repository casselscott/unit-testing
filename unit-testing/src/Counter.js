import React, { useState } from 'react';
import { BiColor, BiMinusCircle } from 'react-icons/bi';
import { BiPlusCircle } from 'react-icons/bi';

const Counter = () => {
  const [Counter, setCounter] = useState(1);

  const increment = () => {
    setCounter((prevCounter) => prevCounter + 1);
  };

  const decrement = () => {
    setCounter((prevCounter) => prevCounter - 1);
  };

  return (
    <div className="Counter">
      <button data-testid="increase" onClick={increment}>
        <BiPlusCircle className="iconplus" />
      </button>
      <p data-testid="number">{Counter}</p>
      <button data-testid="decrease" onClick={decrement}>
        <BiMinusCircle className="iconminus" />
      </button>
    </div>
  );
};

export default Counter;
