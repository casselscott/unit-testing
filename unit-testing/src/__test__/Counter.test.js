import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Counter from '../Counter';

describe('counter', () => {
  it('should increment counter', () => {
    render(<Counter />);

    //select elements to interact
    const counter = screen.getByTestId('number');
    const IncrementBtn = screen.getByTestId('increase');
    const DecrementBtn = screen.getByTestId('decrease');

    //interact with the elements
    fireEvent.click(IncrementBtn);
    expect(counter).toHaveTextContent('2');
  });

  it('should increment counter', () => {
    render(<Counter />);

    //select elements to interact
    const counter = screen.getByTestId('number');
    const DecrementBtn = screen.getByTestId('decrease');

    //interact with the elements
    fireEvent.click(DecrementBtn);
    expect(counter).toHaveTextContent('0');
  });
});

describe('counter snapshots', () => {
  it('should it match DOM Snapshot', () => {
    const data = renderer.create(<Counter />).toJSON();
    expect(data).toMatchSnapshot();
  });
});
