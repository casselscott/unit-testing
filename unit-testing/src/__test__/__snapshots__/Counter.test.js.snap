// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter snapshots should it match DOM Snapshot 1`] = `
<div
  className="Counter"
>
  <button
    data-testid="increase"
    onClick={[Function]}
  >
    <svg
      className="iconplus"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4z"
      />
      <path
        d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
      />
    </svg>
  </button>
  <p
    data-testid="number"
  >
    1
  </p>
  <button
    data-testid="decrease"
    onClick={[Function]}
  >
    <svg
      className="iconminus"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 11h10v2H7z"
      />
      <path
        d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
      />
    </svg>
  </button>
</div>
`;
